Index: MTA10/core/CCore.cpp
===================================================================
--- MTA10/core/CCore.cpp	(revision 2469)
+++ MTA10/core/CCore.cpp	(working copy)
@@ -1206,7 +1206,7 @@
     m_pCommands->Add ( "quit",              "exits the application",            CCommandFuncs::Exit );
     m_pCommands->Add ( "ver",               "shows the version",                CCommandFuncs::Ver );
     m_pCommands->Add ( "time",              "shows the time",                   CCommandFuncs::Time );
-    m_pCommands->Add ( "hud",               "shows the hud",                    CCommandFuncs::HUD );
+    m_pCommands->Add ( "showhud",           "shows the hud",                    CCommandFuncs::HUD );
     m_pCommands->Add ( "binds",             "shows all the binds",              CCommandFuncs::Binds );
 
 #if 0
